### PR TITLE
[Settings] Import Site Editor components from node_modules

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Related to https://github.com/woocommerce/woocommerce/issues/52520

This pull request gathers WP Core's Site Editor components directly from the NPM module. This is a similar tactic used by Customize Your Store.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Turn on the feature flag `settings`, Tools > WCA Test Helper > Features.
2. Visit the new Settings, WooCommerce > Settings. You should a message that Gutenberg must be present.
8. Since we are unlocking the same packages from WP used in the Customizer, we need to make sure the Customizer still functions, WooCommerce > Home > Task List > Customize your store
9. Install/Activate Gutenberg, see the new Settings screen load with a Sidebar and placeholder content
10. Check that Customizer is still functioning

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/changelog/52655-update-settings-gutenberg-components-edit-site [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Remove exported Gutenberg components from `@woocommerce/product-editor`
```
Comment (chihsuan) [inline plugins/woocommerce/changelog/52655-update-settings-gutenberg-components-edit-site]: Is this changelog necessary? If so, should it match changes with ./plugin/woocommerce? e.g. `plugins/woocommerce/package.json`

File: packages/js/settings-editor/src/layout.tsx [hunk 2]
```diff
@@ -22,6 +20,17 @@ import {
 	// @ts-expect-error missing type.
 	__unstableAnimatePresence as AnimatePresence,
 } from '@wordpress/components';
+// eslint-disable-next-line @woocommerce/dependency-group
```
Comment (chihsuan) [inline packages/js/settings-editor/src/layout.tsx:23]: This's looking good. We can also consider disabling the dependency group lint rule for these imports and re-enable it after the import.

```js
/* eslint-disable @woocommerce/dependency-group */

// import...

/* eslint-enable @woocommerce/dependency-group */
```

File: packages/js/settings-editor/changelog/52655-update-settings-gutenberg-components-edit-site [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Remove exported Gutenberg components from `@woocommerce/product-editor`
```
Comment (chihsuan) [inline packages/js/settings-editor/changelog/52655-update-settings-gutenberg-components-edit-site:4]:
```suggestion
Import Site Editor components from node_modules
```

File: packages/js/product-editor/src/lock-unlock.ts [hunk 4]
```diff
@@ -3,19 +3,8 @@
...
-	? __dangerousOptInToUnstableAPIsOnlyForCoreModules(
-			'I acknowledge private features are not for use in themes or plugins and doing so will break in the next version of WordPress.',
-			'@wordpress/edit-site'
-	  )
-	: { lock: noop, unlock: noop };
-
+const { lock, unlock } = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
+	'I acknowledge private features are not for use in themes or plugins and doing so will break in the next version of WordPress.',
+	'@wordpress/edit-site'
+);
```
Comment (chihsuan) [inline packages/js/product-editor/src/lock-unlock.ts:9]: For other reviewers, this just reverts the previous change in https://github.com/woocommerce/woocommerce/pull/52501.

**Review by chihsuan — APPROVED** [review]
Tested well. Pre-approving! 👍

File: plugins/woocommerce/changelog/52655-update-settings-gutenberg-components-edit-site (hunk 1 above)
Comment (psealock) [reply plugins/woocommerce/changelog/52655-update-settings-gutenberg-components-edit-site]: Good call, changed this to a changelog comment instead in https://github.com/woocommerce/woocommerce/pull/52655/commits/135f750863c2715ca682740be35911e105c39543

File: packages/js/settings-editor/src/layout.tsx (hunk 2 above)
Comment (psealock) [reply packages/js/settings-editor/src/layout.tsx:23]: 👍🏽  Done in https://github.com/woocommerce/woocommerce/pull/52655/commits/7e5792fa0898f9f585fd286cb5c9017307f97c87

**Comment by psealock** [comment]
No functionality introduced, so I will merge with just one review

**Pull request merged**